### PR TITLE
fix(react-sdk): improve Zod compatibility and schema test coverage

### PR DESCRIPTION
**Pull request by lachieh**
## Summary

- Simplify Zod peer dependency to support `^3.25.76 || ~4.0.0` (drops 4.1 which caused TypeScript issues)
- Remove internal `zodInternalAlias` workaround that was no longer needed
- Fix `isStandardSchema` to use duck typing instead of Zod 4 validation (avoids cross-version compatibility issues)
- Improve Zod 4 function schema detection and arg/return extraction
- Add comprehensive test coverage for schema utilities

### Zod Compatibility
- Changed peer dependency from `^3.25.76 || ^4.1.0` to `^3.25.76 || ~4.0.0`
- Removed `zodInternalAlias` dev dependency that was used as a workaround
- Made `zod` a required (not optional) peer dependency

### Schema Detection Improvements
- Rewrote `isStandardSchema` to use duck typing instead of Zod 4's `z.object().safeParse()`, which was throwing errors when validating Zod 3 schemas
- Updated `isZod4FunctionSchema` to properly detect v4.0.x function schemas via `def.type === "function"`
- Fixed `getZodFunctionArgs` and `getZodFunctionReturns` to extract args/returns from Zod 4 function schemas

### Test Coverage
- Added `standard-schema.test.ts` with comprehensive coverage (100% statements/branches)
- Added tests for `validate.ts` covering JSON Schema conditional keywords (if/then/else, not, oneOf), prefixItems, and edge cases
- Added tests for `schema.ts` covering toolSchema with JSON Schema tuples and edge cases
- Updated `zod.test.ts` to verify Zod 4 function schema extraction works correctly

## Test Plan
- [x] All 773 tests pass
- [x] Type checking passes
- [x] Lint passes
- [x] Coverage improved: `standard-schema.ts` 88.88% → 100%, `validate.ts` 82.75% → 96.55%

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: react-sdk/src/schema/zod.ts [hunk 1]
```diff
@@ -147,17 +176,45 @@ export function isZod4Schema(schema: unknown): schema is $ZodType {
...
+        "output" in def &&
+        typeof def.output === "object" &&
+        def.output !== null &&
+        "~standard" in def.output;
+      return inputIsZod || outputIsZod;
+    }
+  }
+
+  return false;
 }
```
[the remark this answers is not in the document]
Comment (lachieh) [reply react-sdk/src/schema/zod.ts:221]: This already happens on `L185`...

File: react-sdk/src/schema/schema.test.ts [hunk 2]
```diff
@@ -411,6 +411,97 @@ describe("schema utilities", () => {
...
+          description: "Test tool",
+          tool: jest.fn(),
+          toolSchema: { notASchema: true } as any,
+        };
+
+        // This throws because it's not a Zod function schema
+        expect(() => getParametersFromToolSchema(tool)).toThrow(
+          "Unable to determine parameters from zod function schema",
+        );
+      });
```
[the remark this answers is not in the document]
Comment (lachieh) [reply react-sdk/src/schema/schema.test.ts]: @CharlieHelps yes please

File: react-sdk/src/schema/zod.ts [hunk 3]
```diff
@@ -133,12 +137,37 @@ export function isZod3Schema(schema: unknown): schema is ZodType {
...
+  // This is a special case where _def exists but it's still a v4 schema
+  if ("_def" in schema && "def" in schema) {
+    const def = (schema as { def: unknown }).def;
+    if (def && typeof def === "object" && "type" in def) {
+      return def.type === "function";
+    }
+  }
+
+  return false;
 }
```
[the remark this answers is not in the document]
Comment (lachieh) [reply react-sdk/src/schema/zod.ts:172]: I mean, `"string"` is lowercase string literal so any other checks are irrelevant. @CharlieHelps, update the comment, not the test.

**Review by alecf — APPROVED** [review]
